src/components/core/pivotTable/agGridDataSourceUtils.ts

```diff
@@ -60,17 +60,20 @@ export const wrapGetPageWithCaching = (getPage: IGetPage): IGetPage => {
         };
 
         if (firstCachedPageRequest && isEqual(firstCachedPageRequest.requestParams, requestParams)) {
-            return firstCachedPageRequest.response;
+            const result = await firstCachedPageRequest.resultPromise;
+            return result;
         }
 
-        const execution = await getPage(resultSpec, limit, offset);
+        const resultPromise = getPage(resultSpec, limit, offset);
+
         if (firstCachedPageRequest === null) {
             firstCachedPageRequest = {
                 requestParams,
-                response: execution,
+                resultPromise,
             };
         }
 
+        const execution = await resultPromise;
```

Comment:
Should be here just `return await resultPromise`? :)

Reply:
It breaks the no-return-await eslint rule (https://eslint.org/docs/rules/no-return-await)

Reply:
Oh, I see. Thanks! :)